libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -178,24 +173,57 @@ AP_WindVane *AP_WindVane::get_singleton()
 // return true if wind vane is enabled
 bool AP_WindVane::enabled() const
 {
-    return (_type != WINDVANE_NONE);
+    return _direction_type != WINDVANE_NONE;
 }
 
 // Initialize the Wind Vane object and prepare it for use
 void AP_WindVane::init()
 {
-    // a pin for reading the Wind Vane voltage
-    dir_analog_source = hal.analogin->channel(_dir_analog_pin);
-
-    // pins for ModernDevice rev p wind sensor
-    speed_analog_source = hal.analogin->channel(ANALOG_INPUT_NONE);
-    speed_temp_analog_source = hal.analogin->channel(ANALOG_INPUT_NONE);
+    // wind direction
```

Comment:
often we have a check at the top to ensure init() isn't called more than once.  i.e. if _direction_driver != nullptr; return.

Reply:
fixed